resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -306,9 +318,14 @@ private[yarn] class YarnAllocator(
       s"executorsStarting: ${numExecutorsStarting.get}")
 
     if (missing > 0) {
-      logInfo(s"Will request $missing executor container(s), each with " +
-        s"${resource.getVirtualCores} core(s) and " +
-        s"${resource.getMemory} MB memory (including $memoryOverhead MB of overhead)")
+      var requestContainerMessage = s"Will request $missing executor container(s), each with " +
+          s"${resource.getVirtualCores} core(s) and " +
+          s"${resource.getMemory} MB memory (including $memoryOverhead MB of overhead) "
+      if (ResourceTypeHelper.isYarnResourceTypesAvailable()) {
```

Comment:
I would remove this condition. You already have it in `ResourceTypeHelper.getCustomResourcesAsStrings`. Better to write every method such that it won't crash from reflection than to try to force the developer to remember what methods can throw reflection errors.

Reply:
You made a very good point with this, fixed.